ls1intum/Artemis
# `Development`: Replace deprecated Angular component factory

### Checklist
#### General


#### Server
#### Changes affecting Programming Exercises

### Motivation and Context



The [ComponentFactoryResolver](https://angular.io/api/core/ComponentFactoryResolver) in Angular was deprecated. 

### Description


I therefore replaced the code using [ViewContainerRef](https://angular.io/api/core/ViewContainerRef), like [suggested](https://stackoverflow.com/q/70946038).

### Steps for Testing

Prerequisites:
- 1 Instructor/Student
- 1 Programming Exercise with Tasks

1. Log in to Artemis
2. Navigate to the Programming Exercise
4. Ensure that the task status of instructions are rendered correctly (also try different situations, e.g., submitting code)

### Review Progress





#### Code Review
#### Manual Tests
